pupil-spa/src/app/check/check.component.ts

```diff
@@ -130,12 +135,16 @@ export class CheckComponent implements OnInit {
         // Show the question screen
         this.isWarmUp = false;
         const matches = /^Q(\d+)$/.exec(stateDesc);
+        this.registerInputService.flush();
         this.question = this.questionService.getQuestion(parseInt(matches[ 1 ], 10));
+        this.registerInputService.initialise();
         this.viewState = 'question';
         break;
       }
       case(/^complete$/).test(stateDesc):
         // Show the check complete screen
+        this.registerInputService.flush();
```

Comment:
Shouldn't `->flush()` be called at the very end, before the `CheckComplete` audit event?

Reply:
@jon-shipley Not sure I am following, isn't that what is happening currently?

Reply:
Okay - that's fine - didn't realise it's a different service